src/parachain/loans.ts

```diff
@@ -730,27 +737,31 @@ export class DefaultLoansAPI implements LoansAPI {
         currentBlock: number
     ): Promise<MonetaryAmount<CurrencyExt>> {
         const [latestSupplyIndex, rewardSupplierIndex, lendTokenRawBalance] = await Promise.all([
-            this._getLatestSupplyIndex(underlyingCurrencyId, lendTokenId, currentBlock),
+            this._getLatestSupplyIndex(underlyingCurrencyId, lendTokenId, currentBlock, rewardCurrency),
             this.api.query.loans.rewardSupplierIndex(underlyingCurrencyId, accountId),
             this.api.query.tokens.accounts(accountId, lendTokenId),
         ]);
-        const supplierIndex = decodeFixedPointType(rewardSupplierIndex);
+        const supplierIndex = Big(rewardSupplierIndex.toString());
         const lendTokenBalance = Big(lendTokenRawBalance.free.toString()).add(lendTokenRawBalance.reserved.toString());
         const deltaIndex = latestSupplyIndex.sub(supplierIndex);
 
-        const accruedRewardInPlanck = deltaIndex.mul(lendTokenBalance);
+        const accruedRewardInPlanck = deltaIndex.mul(lendTokenBalance).div(10 ** rewardCurrency.decimals);
```

Comment:
Not blocking: Is the variable name `accruedRewardInPlanck` still accurate?
It's not immediately clear to me if we're simply dividing by 10**decimals because the index now has been multiplied by it, or whether this normalizes the value.

Reply:
This is still accurate because index is multiplied by 10 ** decimals, so to compute the reward in Planck we have to divide it